docs/merkletree/treetex/main.go

```diff
@@ -318,6 +326,32 @@ func main() {
 	flag.Parse()
 	height := uint(bits.Len64(*treeSize-1)) + 1
 
+	var innerNodeText = func(id compact.NodeID) string {
+		if id.Level == 0 {
+			panic("attempting to use inner node text func for leaf text")
+		}
+		return fmt.Sprintf("%d.%d", id.Level, id.Index)
+	}
+
+	var ldf nodeTextFunc = func(id compact.NodeID) string {
+		if id.Level == 0 {
+			return fmt.Sprintf("%d", id.Index)
+		}
+		return innerNodeText(id)
+	}
+
+	if len(*leafData) > 0 {
+		leaves := strings.Split(*leafData, ",")
```

Comment:
How about pre-computing `leaves` for the other case? Then the node function can be shared, and you can merge all 3 function into 1 nice little function.

Reply:
My thinking was that if you're specifying `--leaf_data` then there aren't going to be that many leaves, so keeping them in a slice was fine, but otherwise you can set `--tree_size` to 100,000 (and it'll still work because of the "perfect tree collapsing"), but it'll only need a handful of _actual_ leaves to display.

e.g.:
![image](https://user-images.githubusercontent.com/7648032/69748612-a8746180-1140-11ea-974f-be08fc1067e8.png)

Reply:
SNTBAATM (sounds not too bad at all to me)